dp/cloud/go/services/dp/obsidian/cbsd/handlers.go

```diff
@@ -216,6 +218,9 @@ func getHttpError(err error) error {
 	case codes.NotFound:
 		return echo.NewHTTPError(http.StatusNotFound, err)
 	default:
+		if strings.Contains(fmt.Sprintf("%s", err), duplicateRecordMsg) == true {
```

Comment:
I think we should not rely on error message (which can depend on database implementation), but instead return proper response code, by the servicers `codes.AlreadyExists` especially, since handlers do not talk directly to the database and are not supposed to understand it.
So I would suggest updating also servicers (return proper error code `codes.AlreadyExists`)
and db handlers (return proper error `merrors.ErrAlreadyExists`)